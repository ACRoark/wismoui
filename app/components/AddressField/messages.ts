import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.address-field';

export default defineMessages({
  addressFieldLabel: {
    id: `${scope}.address-field-label`,
    defaultMessage: 'Address',
  },
});
