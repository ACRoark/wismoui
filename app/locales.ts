export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
