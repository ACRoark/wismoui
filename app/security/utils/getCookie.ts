const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
