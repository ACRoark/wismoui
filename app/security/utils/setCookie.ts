interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
