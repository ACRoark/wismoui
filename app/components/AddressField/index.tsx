import React from 'react';

import Address from 'components/Address';
import BaseFormField from 'components/BaseField';

import IAddress from 'types/IAddress';

import messages from './messages';

interface IAddressFieldProps {
  address: IAddress;
}

const AddressField: React.FC<IAddressFieldProps> = (props: IAddressFieldProps): React.ReactElement => {
  const { address } = props;

  return (
    <BaseFormField label={messages.addressFieldLabel}>
      <Address address={address} />
    </BaseFormField>
  );
};

export default AddressField;
