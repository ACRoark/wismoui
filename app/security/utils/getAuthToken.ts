import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
