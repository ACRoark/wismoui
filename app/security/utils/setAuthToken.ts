import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
