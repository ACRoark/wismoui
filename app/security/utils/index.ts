import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
