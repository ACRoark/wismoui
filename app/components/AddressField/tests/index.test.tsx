import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import IAddress from 'types/IAddress';

import AddressField from '..';

describe('<AddressField />', (): void => {
  let address: IAddress;
  const languages = Object.keys(SUPPORTED_LOCALES);

  beforeEach((): void => {
    address = {
      city: 'Anytown',
      line1: '1234 That Street',
      line2: null,
      state: 'MI',
      zip: '12345-1234',
    };
  });

  languages.forEach((language: string): void => {
      it(`should not log errors in console when rendered in ${language}`, (): void => {
        const spy = jest.spyOn(global.console, 'error');

        renderWithIntl(<AddressField address={address} />, language);

        expect(spy).not.toHaveBeenCalled();
      });

      it(`should render correctly when a valid address is passed in and the current language is ${language}`, (): void => {
        const tree = createSnapshotWithIntl(<AddressField address={address} />, language);

        expect(tree).toMatchSnapshot();
      });
  });
});
