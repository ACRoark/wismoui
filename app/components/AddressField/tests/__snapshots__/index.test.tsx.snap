// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressField /> should render correctly when a valid address is passed in and the current language is en 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Address
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <address
      className="dte-wismo-address"
    >
      1234 That Street
Anytown, MI 12345
    </address>
  </span>
</div>
`;

exports[`<AddressField /> should render correctly when a valid address is passed in and the current language is es 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Direcci√≥n
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <address
      className="dte-wismo-address"
    >
      1234 That Street
Anytown, MI 12345
    </address>
  </span>
</div>
`;
